components/ai_chat/resources/untrusted_conversation_frame/components/assistant_response/tool_event.tsx

```diff
@@ -75,14 +83,20 @@ function ToolEventContent(
   }
 
   // If we have a tool-specific component, use it, otherwise use the default
-  let component: ToolComponent | null = null
-
   if (toolUseEvent.toolName === Mojom.USER_CHOICE_TOOL_NAME) {
-    component = ToolEventContentUserChoice
+    return <ToolEventContentUserChoice
```

Comment:
Why did we make this change? Looks like we're going to have to pass all these props to all tools, so the previous pattern had less boilerplate?

Reply:
I think this was because React was complaining about using different hooks in the components on each render, maybe because it doesn't treat them as actual Components in the way I had it previously?
I have some tools upcoming that each use different hooks and that's where I saw the error.
Happy to take suggestions but on this PR I was very focused on the tab management tool itself and the browser tests, and not so much on the UI.

Reply:
Ah, maybe you want 

```
React.createElement(component, props)

// instead of
component(props)
```